packages/rich-text/src/component/index.js

```diff
@@ -258,18 +258,32 @@ class RichText extends Component {
 		}
 
 		if ( onPaste ) {
-			// Only process file if no HTML is present.
-			// Note: a pasted file may have the URL as plain text.
-			const image = find( [ ...items, ...files ], ( { type } ) =>
-				/^image\/(?:jpe?g|png|gif)$/.test( type )
```

Comment:
Moved this check to the RichText for block implementation (which only handles images), so we generally pass any files.